File: app/src/main/java/com/custu/project/walktogether/QuestionEightActivity.java
package com.custu.project.walktogether;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.TextView;

import com.custu.project.project.walktogether.R;
import com.custu.project.walktogether.adapter.HintAdapter;
import com.custu.project.walktogether.data.Evaluation.Question;
import com.custu.project.walktogether.model.EvaluationModel;
import com.custu.project.walktogether.util.BasicActivity;
import com.custu.project.walktogether.util.StoreAnswerTmse;

import java.util.ArrayList;
import java.util.List;

public class QuestionEightActivity extends AppCompatActivity implements BasicActivity,View.OnClickListener {
    private Spinner answerSpinnerOne;
    private Spinner answerSpinnerTwo;
    private Spinner answerSpinnerThree;
    private Spinner answerSpinnerFour;
    private Spinner answerSpinnerFive;
    private List<String> answerArray = new ArrayList<String>();
    private Button nextBtn;
    private Question question;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.question_eight);
        getData();
        setUI();
        setListener();
        createSpinnerData();
        setAdapter();
    }

    @Override
    public void initValue() {

    }

    public void setUI() {
        answerSpinnerOne = (Spinner) findViewById(R.id.answer_day_one);
        answerSpinnerTwo = (Spinner) findViewById(R.id.answer_day_two);
        answerSpinnerThree = (Spinner) findViewById(R.id.answer_day_three);
        answerSpinnerFour = (Spinner) findViewById(R.id.answer_day_four);
        answerSpinnerFive = (Spinner) findViewById(R.id.answer_day_five);
        nextBtn = (Button) findViewById(R.id.next);
        TextView title = findViewById(R.id.title);
        TextView description = findViewById(R.id.description);
        title.setText(question.getTitle());
        description.setText(question.getDescription());
    }

    @Override
    public void getData() {
        question = EvaluationModel.getInstance().getEvaluationByNumber("8", QuestionEightActivity.this).getQuestion();
    }

    private boolean isCorrect() {
        boolean isCorrect = true;
        if (!answerSpinnerOne.getSelectedItem().toString().equalsIgnoreCase(answerArray.get(4)))
            isCorrect = false;

        if (!answerSpinnerTwo.getSelectedItem().toString().equalsIgnoreCase(answerArray.get(5)))
            isCorrect = false;

        if (!answerSpinnerThree.getSelectedItem().toString().equalsIgnoreCase(answerArray.get(2)))
            isCorrect = false;

        if (!answerSpinnerFour.getSelectedItem().toString().equalsIgnoreCase(answerArray.get(3)))
            isCorrect = false;

        if (!answerSpinnerFive.getSelectedItem().toString().equalsIgnoreCase(answerArray.get(0)))
            isCorrect = false;

        Log.d("isCorrect: ", "isCorrect: "+isCorrect);
        return isCorrect;
    }

    @Override
    public void initProgressDialog() {

    }

    public void setAdapter() {
        HintAdapter adapter = new HintAdapter(QuestionEightActivity.this, answerArray, android.R.layout.simple_spinner_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);

        adapter.setDropDownViewResource(R.layout.spinner_layout);
        answerSpinnerOne.setAdapter(adapter);
        answerSpinnerTwo.setAdapter(adapter);
        answerSpinnerThree.setAdapter(adapter);
        answerSpinnerFour.setAdapter(adapter);
        answerSpinnerFive.setAdapter(adapter);

        answerSpinnerOne.setSelection(adapter.getCount());
        answerSpinnerTwo.setSelection(adapter.getCount());
        answerSpinnerThree.setSelection(adapter.getCount());
        answerSpinnerFour.setSelection(adapter.getCount());
        answerSpinnerFive.setSelection(adapter.getCount());
    }

    private void createSpinnerData() {
        answerArray.add("วันจันทร์");
        answerArray.add("วันอาทิตย์");
        answerArray.add("วันพุธ");
        answerArray.add("วันอังคาร");
        answerArray.add("วันศุกร์");
        answerArray.add("วันพฤหัสบดี");
        answerArray.add("วันเสาร์");
        answerArray.add("วัน...");
    }
    private void setListener() {
        nextBtn.setOnClickListener(this);

    }
    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.next: {
                StoreAnswerTmse.getInstance().storeAnswer("no8", question.getId(), String.valueOf(isCorrect()));
                Intent intent = new Intent(QuestionEightActivity.this, QuestionNineActivity.class);
                startActivity(intent);
            }
        }
    }
}

File: app/src/main/java/com/custu/project/walktogether/controller/tmse/QuestionEightActivity.kt
package com.custu.project.walktogether.controller.tmse

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.CountDownTimer
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.ProgressBar
import android.widget.Spinner
import android.widget.TextView

import com.custu.project.project.walktogether.R
import com.custu.project.walktogether.adapter.HintAdapter
import com.custu.project.walktogether.data.Evaluation.Question
import com.custu.project.walktogether.model.EvaluationModel
import com.custu.project.walktogether.util.BasicActivity
import com.custu.project.walktogether.util.ConfigService
import com.custu.project.walktogether.util.DialogUtil
import com.custu.project.walktogether.util.StoreAnswerTmse

import java.util.ArrayList

import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper

class QuestionEightActivity : AppCompatActivity(), BasicActivity, View.OnClickListener {
    private var answerSpinnerOne: Spinner? = null
    private var answerSpinnerTwo: Spinner? = null
    private var answerSpinnerThree: Spinner? = null
    private var answerSpinnerFour: Spinner? = null
    private var answerSpinnerFive: Spinner? = null
    private val answerArray = ArrayList<String>()
    private var nextBtn: Button? = null
    private var question: Question? = null

    private var countDownTimer: CountDownTimer? = null

    private val isCorrect: Boolean
        get() {
            var isCorrect = true
            if (!answerSpinnerOne!!.selectedItem.toString().equals(answerArray[4], ignoreCase = true))
                isCorrect = false

            if (!answerSpinnerTwo!!.selectedItem.toString().equals(answerArray[5], ignoreCase = true))
                isCorrect = false

            if (!answerSpinnerThree!!.selectedItem.toString().equals(answerArray[2], ignoreCase = true))
                isCorrect = false

            if (!answerSpinnerFour!!.selectedItem.toString().equals(answerArray[3], ignoreCase = true))
                isCorrect = false

            if (!answerSpinnerFive!!.selectedItem.toString().equals(answerArray[0], ignoreCase = true))
                isCorrect = false

            Log.d("isCorrect: ", "isCorrect: $isCorrect")
            return isCorrect
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.question_eight)
        getData()
        setUI()
        setListener()
        createSpinnerData()
        setAdapter()
        countDownTime()
    }

    private fun countDownTime() {
        val timeInterval = ConfigService.TIME_INTERVAL
        val time = intArrayOf(31)
        val progress: ProgressBar = findViewById(R.id.progress)
        progress.max = time[0]
        progress.progress = time[0]
        countDownTimer = object : CountDownTimer(timeInterval, 1000) {
            override fun onTick(millisUntilFinished: Long) {
                progress.progress = --time[0]
            }

            override fun onFinish() {
                progress.progress = 0
                countDownTimer!!.cancel()
                StoreAnswerTmse.getInstance().storeAnswer("no8", question!!.id, "")
                val intent = Intent(this@QuestionEightActivity, QuestionNineActivity::class.java)
                startActivity(intent)
            }
        }.start()
    }

    override fun onBackPressed() {
        countDownTimer!!.cancel()
        DialogUtil.getInstance().showDialogExitEvaluation(this)
    }

    override fun initValue() {

    }

    @SuppressLint("SetTextI18n")
    override fun setUI() {
        answerSpinnerOne = findViewById<View>(R.id.answer_day_one) as Spinner
        answerSpinnerTwo = findViewById<View>(R.id.answer_day_two) as Spinner
        answerSpinnerThree = findViewById<View>(R.id.answer_day_three) as Spinner
        answerSpinnerFour = findViewById<View>(R.id.answer_day_four) as Spinner
        answerSpinnerFive = findViewById<View>(R.id.answer_day_five) as Spinner
        nextBtn = findViewById<View>(R.id.next) as Button
        val title = findViewById<TextView>(R.id.title)
        val description = findViewById<TextView>(R.id.description)
        title.text = "(8) " + question!!.title
        description.text = question!!.description
    }

    override fun getData() {
        question = EvaluationModel.getInstance().getEvaluationByNumber("8", this@QuestionEightActivity)!!.question
    }

    override fun initProgressDialog() {

    }

    fun setAdapter() {
        val adapter = HintAdapter(this@QuestionEightActivity, answerArray, android.R.layout.simple_spinner_item)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)

        adapter.setDropDownViewResource(R.layout.spinner_layout)
        answerSpinnerOne!!.adapter = adapter
        answerSpinnerTwo!!.adapter = adapter
        answerSpinnerThree!!.adapter = adapter
        answerSpinnerFour!!.adapter = adapter
        answerSpinnerFive!!.adapter = adapter

        answerSpinnerOne!!.setSelection(adapter.count)
        answerSpinnerTwo!!.setSelection(adapter.count)
        answerSpinnerThree!!.setSelection(adapter.count)
        answerSpinnerFour!!.setSelection(adapter.count)
        answerSpinnerFive!!.setSelection(adapter.count)
    }

    private fun createSpinnerData() {
        answerArray.add("วันจันทร์")
        answerArray.add("วันอาทิตย์")
        answerArray.add("วันพุธ")
        answerArray.add("วันอังคาร")
        answerArray.add("วันศุกร์")
        answerArray.add("วันพฤหัสบดี")
        answerArray.add("วันเสาร์")
        answerArray.add("วัน...")
    }

    private fun setListener() {
        nextBtn!!.setOnClickListener(this)

    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.next -> {
                countDownTimer!!.cancel()
                StoreAnswerTmse.getInstance().storeAnswer("no8", question!!.id, isCorrect.toString())
                val intent = Intent(this@QuestionEightActivity, QuestionNineActivity::class.java)
                startActivity(intent)
            }
        }
    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(base))
    }
}



